include/graft_utility.hpp

```diff
@@ -16,13 +20,95 @@ namespace graft
             mutable std::mutex m;
             std::shared_ptr<T> data;
             std::unique_ptr<node>  next;
+            ch::seconds ttl;
+            ch::seconds expires;
+
+            node() : next(), expires(ch::seconds::max()), ttl(ch::seconds(0)) {}
+
+            node(T const& value)
+                : expires(ch::seconds::max())
+                , ttl(ch::seconds(0))
+                , data(std::make_shared<T>(value)) {}
+
+            node(T const& value, std::chrono::seconds ttl)
+                : ttl(ttl), data(std::make_shared<T>(value)) { update_time(); }
+
+            void update_time()
+            {
+                expires = (ttl == ch::seconds(0)) ?
+                    ch::seconds::max() : ch::time_point_cast<ch::seconds>(
+                        ch::steady_clock::now()
+                    ).time_since_epoch() + ttl;
+            }
+
+            bool expired()
+            {
+                return expires <=
+                    ch::time_point_cast<ch::seconds>(
+                        ch::steady_clock::now()
+                    ).time_since_epoch();
+            }
 
-            node() : next() {}
-            node(T const& value) : data(std::make_shared<T>(value)) {}
+            bool operator <=(const node& rhs) const
+            {
+                return expires <= rhs.expires;
+            }
         };
 
         node head;
 
+        void fixup_tail(node *current)
+        {
+            std::unique_lock<std::mutex> lk(current->m);
+            while (node* const next = current->next.get())
+            {
+                std::unique_lock<std::mutex> next_lk(next->m);
+
+                if (next->expired())
+                {
+                    std::unique_ptr<node> old_next = std::move(current->next);
+                    current->next = std::move(next->next);
```

Comment:
Looks like instead of
current->next = std::move(next->next);
you can do 
current->next = nullptr;
because it is expected that if next expired it means whole tail after the next is expired